cxmon/src/mon_cmd.cpp

```diff
@@ -303,6 +307,288 @@ void binary_dump(void)
+ */
+void break_point_enable(void)
+{
+	uintptr index;
+
+	if (mon_token == T_END ||
+			!mon_expression(&index)) {
+		fprintf(monerr, "Expect index number of break point in hexadecimal.\n");
+		return;
+	}
+
+	if (mon_token != T_END) {
+		mon_error("Too many arguments");
+		return;
+	}
+
+	if (index > disabled_break_points.size()) {
+		mon_error("Illegal Index Number!");
+		return;
+	}
+
+	BREAK_POINT_SET::iterator it;
+
+	if (0 == index) {
+		for (it = disabled_break_points.begin(); it != disabled_break_points.end(); it++)
+			active_break_points.insert(*it);
+		disabled_break_points.clear();
+		printf("Enabled all break points!\n");
+		return;
+	}
+
+	int pos = 1;
+	for (it = disabled_break_points.begin(); it != disabled_break_points.end(); it++)
+		if (pos++ == index)
+			break;
+	// Add to active break points
+	printf("Disabled break point %4x at address %08lx\n", index, *it);
+	active_break_points.insert(*it);
+	// Remove break point
+	disabled_break_points.erase(it);
+}
+
+
+/*
+ * List all Active Break Points
+ */
+void break_point_info(void)
+{
+	if (mon_token != T_END) {
+		mon_error("Too many arguments");
+		return;
+	}
+
+	BREAK_POINT_SET::iterator it;
+	int pos;
+
+	if (!active_break_points.empty()) {
+		pos = 1;
+		printf(STR_ACTIVE_BREAK_POINTS);
+		for (it = active_break_points.begin(); it != active_break_points.end(); it++)
+			printf("\tBreak point %4x at address %08lx\n", pos++, *it);
+	}
+
+	if (!disabled_break_points.empty()) {
+		putchar('\n');
+		printf(STR_DISABLED_BREAK_POINTS);
+		pos = 1;
+		for (it = disabled_break_points.begin(); it != disabled_break_points.end(); it++)
+			printf("\tBreak point %4x at address %08lx\n", pos++, *it);
+	}
+}
+
+
+/*
+ * Save all Active Break Points to a file
+ */
+void break_point_save(void)
+{
+	FILE *file;
+
+	if (mon_token == T_END) {
+		mon_error("Missing file name");
+		return;
+	}
+	if (mon_token != T_STRING) {
+		mon_error("'\"' around file name expected");
+		return;
+	}
+	mon_get_token();
+	if (mon_token != T_END) {
+		mon_error("Too many arguments");
+		return;
+	}
+
+	if (!(file = fopen(mon_string, "w")))
+		mon_error("Unable to create file");
+	else {
+		BREAK_POINT_SET::iterator it;
+
+		fprintf(file, STR_ACTIVE_BREAK_POINTS);
+		for (it = active_break_points.begin(); it != active_break_points.end(); it++)
+			fprintf(file, "%x\n", *it);
+
+		fprintf(file, STR_DISABLED_BREAK_POINTS);
+		for (it = disabled_break_points.begin(); it != disabled_break_points.end(); it++)
+				fprintf(file, "%x\n", *it);
+
+		fclose(file);
+	}
+}
+
+
+/*
+ * Load Break Point from a file
+ */
+void break_point_load(void)
+{
+	FILE *file;
+
+	if (mon_token == T_END) {
+		mon_error("Missing file name");
+		return;
+	}
+	if (mon_token != T_STRING) {
+		mon_error("'\"' around file name expected");
+		return;
+	}
+	mon_get_token();
+	if (mon_token != T_END) {
+		mon_error("Too many arguments");
+		return;
+	}
+
+	if (!(file = fopen(mon_string, "r")))
+		mon_error("Unable to create file");
+	else{
```

Comment:
Can you go over your changed lines and use a consistent style? i.e. space before { and between "if" and "(".

Reply:
Adding a space between `if` and `(` and between `(` and `{` is exactly style in the original mon_cmd.cpp. I just follow what it is already there.

Reply:
I see what you meant now. Fixed it in 597ff06